source/fundamentals/enterprise-auth.txt

```diff
@@ -1,27 +1,186 @@
+  principal name. A sample username is *myuser@KERBEROS.EXAMPLE.COM*.
+- ``password``: Your Kerberos user's password.
+- ``hostname``: The network address of your MongoDB server, accessible by your client.
+- ``port``: Your MongoDB server's port number.
+
+.. note::
+
+   Using Kerberos authentication requires the "gssapi" build tag and ``cgo``
+   support during compilation. The default service name for Kerberos is
+   "mongodb". You can define the ``AuthMechanismProperties`` field in
+   the ``Credential`` struct to use a different service name.
+
+The following code shows how you can define a ``Credential`` struct to
+authenticate to Kerberos and create a client with your authentication
+preferences:
+
+.. code-block:: go
+
+   credential := options.Credential{
+           AuthMechanism: "GSSAPI",
+           Username:      "<username>",
+           Password:      "<password>",
+   }
+
+   uri := "mongodb://<hostname>:<port>"
+   clientOpts := options.Client().ApplyURI(uri).SetAuth(credential)
+   
+   client, err := mongo.Connect(context.TODO(), clientOpts)
+
+.. tip::
+
+   You don't need to define a password in your ``Credential`` struct if
+   you store authentication keys in ``keytab`` files. You can initialize
+   a credential cache for authenticating the user principal using the
+   ``kinit`` binary. The following command shows how you can invoke a
+   credential cache for a sample username:
+   
+   .. code-block:: sh
+   
+      kinit myuser@KERBEROS.EXAMPLE.COM
+
+.. _golang-LDAP:
+
+Authenticate with LDAP (PLAIN)
+------------------------------
+
+You can authenticate to a Lightweight Directory Access Protocol (LDAP) server 
+using your directory-server username and password.
+
+.. warning::
+
+   This authentication mechanism sends the password to the server in
+   plaintext, so you should use this mechanism only with TLS connections.
+
+The following example specifies the authentication mechanism using the following
+placeholders:
+
+- ``username``: Your LDAP username
+- ``password``: Your LDAP password
+- ``hostname``: The network address of your MongoDB server, accessible
+  by your client
+- ``port``: Your MongoDB server's port number
+
+The following code shows how you can define a ``Credential`` struct to
+authenticate to LDAP and create a client with your authentication
+preferences:
+
+.. code-block:: go
+
+   credential := options.Credential{
+           AuthMechanism: "PLAIN",
+           Username:      "<username>",
+           Password:      "<password>",
+	}
+
+   uri := "mongodb://<hostname>:<port>"
+   clientOpts := options.Client().ApplyURI(uri).SetAuth(credential)
+   
+   client, err := mongo.Connect(context.TODO(), clientOpts)
+
+.. tip::
+
+   The method refers to PLAIN instead of LDAP since it
+   authenticates using the PLAIN Simple Authentication and Security Layer
+   (SASL) defined in `RFC-4616 <https://tools.ietf.org/html/rfc4616>`__.
+         
+Configuration Options
```

Comment:
The only additional option that isn't for Kerberos is `AWS_SESSION_TOKEN`. I think I'll just move this content under the Kerberos header.

Reply:
Suggestion: 
I would have expected to see the options prior to the example. Is there a good reason to place them after the example? If so, I think it could be helpful to introduce the subsections in the overview of the top-level section so that the reader can locate the information more quickly (rather than relying on the right-hand On this page nav).